server/routes/admin.js

```diff
@@ -1,20 +1,54 @@
-var express = require('express');
-var router = express.Router();
+const express = require('express');
+const router = express.Router();
 const categoryClient = require('../dbClients/categoriesDB');
 const articleClient = require('../dbClients/articlesDB')
- 
+const ObjectId = require('mongodb').ObjectID;
 
 router.get("/", function (req, res, next) {
   res.send('admin');
 });
 
 router.get('/categories', function (req, res, next) {
   const callback = (error, category) => {
-     res.render("admin-list-categories",{
+    res.render("admin-list-categories", {
       category
-     }) 
+    })
+  }
+  categoryClient.findCategories({},callback);
+});
+
+router.get('/articles', function (req, res, next) {
+  const callback = (error, articles) => {
+    res.render("admin-list-articles", {
+      articles: articles
+    })
+  }
+  articleClient.findArticles({}, callback);
+});
+ 
+router.get('/categories/edit/:categoryId', (req, res) => {
+  let categoryId = req.params.categoryId;
+  const callback = (error, category) => {
+    res.render("admin-edit-category", {
+      category: category[0]
+    })
+  }
+  categoryClient.findCategories({ "_id": ObjectId(categoryId) }, callback);
+})
+router.post('/categories/edit/:categoryId', function (req, res, next) {
+  let categoryId = req.params.categoryId;
+  const query = req.body;
+  const callback = (error, category) => {
+// we use find articles function to show us list of all articles after update
+    const callbacktwo = (error, category) => {
+      res.render("admin-list-categories", {
+        category: category
+      })
     }
-    categoryClient.listCategory(callback);
+    categoryClient.findCategories({}, callbacktwo);
+  }
+  categoryClient.editCategory({ "_id": ObjectId(categoryId) }, query, true, callback);
```

Comment:
if the upsert option is always true, then just let the method in DBClient have it true. No need to pass it as an argument

Reply:
again I don't think you need ObjectId (try though)